src/android/com/megster/cordova/BluetoothSerial.kt

```diff
@@ -26,27 +26,26 @@ class BluetoothSerial : CordovaPlugin() {
     override fun execute(action: String, args: CordovaArgs, callbackContext: CallbackContext): Boolean {
         LOG.d(TAG, "action = $action")
         var validAction = true
+
+        try {
+            enableBluetoothIfNecessary()
+        } catch (e: Exception) {
+            // If we are unable to enable bluetooth
+            // then return false, which will invoke
+            // the error callback
+            Log.e(TAG, "Unable to enable bluetooth: $e")
+            return false
+        }
+
         when (action) {
             CONNECT -> {
-                enableBluetoothIfNecessary()
                 connect(args, callbackContext)
             }
             DISCONNECT -> {
-                try {
-                    BluetoothSerialService.stop()
-                    callbackContext.success()
-                } catch (e: Exception) {
-                    callbackContext.error(e.toString())
-                }
+                disconnect(callbackContext)
             }
             SEND -> {
-                try {
-                    val data: ByteArray = args.getArrayBuffer(0)
-                    BluetoothSerialService.write(data)
-                    callbackContext.success()
-                } catch (e: Exception) {
-                    callbackContext.error(e.toString())
-                }
+                send(args, callbackContext)
```

Comment:
If we are going to follow this pattern, lets move the other actions to this pattern as we touch them.
Why not `this.send(args, callbackContext)`
:P

Reply:
you are right, i should add `this` everywhere!!  i can't believe i missed that :)